stac_ipyleaflet/stac_discovery/requests.py

```diff
@@ -0,0 +1,20 @@
+import requests
+import logging
+from stac_ipyleaflet.constants import REQUEST_TIMEOUT
+
+def make_get_request(url, params=None, timeout=REQUEST_TIMEOUT):
+    """GET Request wrapper to watch for and catch specific errors
+    Args:
+        url (str): HTTP URL to a STAC item
+        params (dict[str, Any]): request parameters
+        timeout: duration for requests to stop waiting for a response after a given number of seconds
+    """
+    try:
+        req = requests.get(url, params=params, timeout=timeout)
+        return req
+    except requests.exceptions.Timeout:
+        # QUESTION: Should we retry?
+        logging.error(f"timeout raised during get request")
+    except requests.exceptions.RequestException as e:
+        logging.error(e)
+    return
```

Comment:
📓 : adding this so we can wrap our requests with a try/catch. If we want to add more exceptions to look for in the future we can easily do so.

❓ : retry logic?

Reply:
I don't think I'm clear on your question here. One thing I'll say is that we may need to replace this request with a STAC call using the `pystac-client` library.

Reply:
I was wondering if we should retry the request if an exception was caught? Not something we would have to decide now and can defer but something to think about